hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -773,7 +773,7 @@ protected void getAllInternal(Set<K> keys, List<Data> dataKeys, List<Object> res
             return Collections.emptyMap();
         }
 
-        Map<Integer, List<Data>> idToKeys = new HashMap<Integer, List<Data>>();
+        Map<Integer, List<Data>> idToKeys = createHashMap(partitionService.getPartitionCount());
```

Comment:
I'm going to revert this as we don't know that there will be enough keys to be distributed on all partitions. Also, we are planning on changing the default partition count to 1999